### ISSUE TITLE
JavaScript Error: ENOENT: no such file or directory

### ISSUE THREAD

**Issue by ourrobin** [Azure/azure-functions-host#5238]
Our JavaScript functions have been running successfully for quite some time, and all of the sudden the HttpTrigger invocation URLs fails with the following error:

`Error: ENOENT: no such file or directory, open 'D:\local\Temp\functions\standby\wwwroot\site\src\index.html'`

The code that is failing is the following:

```
const fs = require('fs');
const Path = require('path');
...
const indexFilePath = Path.resolve(`./site/src/index.html`);
const htmlFile = fs.readFileSync(indexFilePath).toString();
```

The files under our directory `site/src/...` are resources that our function needs to operate (files that have been successfully loaded several times before with the same code) and is deployed successfully to the azure function storage account under file shares `[APP_NAME]-content` in file `data/SitePackages/[timestamp].zip`.

I am able to temporarily solve the issue by redeploying my function app (without any new code changes) with:

`func azure functionapp publish [APP_NAME]`

For some function apps one redeploy solved the issue, for other apps several redeploys were needed.

However, after some time and after some successful invokations the issue starts to occur again.

#### Investigative information

Please provide the following:

- Timestamp: 2019-11-13T08:58:40.899Z
- Function App version (1.0 or 2.0): 2.0
- Invocation ID: 38a47a8a-8cfd-4295-845a-f872e57d65d8
- Region: West US

**Comment by mhoeger** [comment]
Hi @ourrobin - can you try adding an app setting `WEBSITE_USE_PLACEHOLDER` with a value of `0`? This will be a temporary mitigation. We just turned on some performance optimizations and while that will opt you out of it for now, it will let your app keep running as normal until we have a further update here. Thanks!

**Comment by TeraInferno** [comment]
I am seeing the same failures in my TypeScript Azure Functions. In my case, I am using an Event Hub Trigger and  the file in question is a MaxMind DB binary file. 

The code that fails on my end is:
```typescript
import { AzureFunction, Context } from '@azure/functions';
import { Reader } from '@maxmind/geoip2-node';
import ReaderModel from '@maxmind/geoip2-node/dist/src/readerModel';
// ...
let reader: ReaderModel;
// ...
const main: AzureFunction = async function(context: Context, eventHubMessages: any[]): Promise<void> {
// ...
        if (!reader) {
            reader = await Reader.open('DB/GeoLite2-City4.mmdb');
        }
// ...
};
```

The function app is deployed using the [Zip Deployment](https://docs.microsoft.com/en-us/azure/azure-functions/deployment-zip-push) mechanism. Here is how I am doing it:
```bash
npm install
npm run build:production
zip -r ${ZIP_FILE_NAME} . --exclude @.funcignore --exclude .funcignore
az functionapp deployment source config-zip \
    -g ${RESOURCE_GROUP} -n ${APP_NAME} \
    --src ${ZIP_FILE_NAME}
```
I am also setting `WEBSITE_RUN_FROM_PACKAGE` to `1` in my application settings as well.

**Comment by ourrobin** [comment]
Adding flag `WEBSITE_USE_PLACEHOLDER` with value `0` does indeed fix the issue. Thanks! :+1:

**Comment by mhoeger** [comment]
Cc: @pragnagopa

**Comment by mhoeger** [comment]
Re-opening issue as the fix not yet deployed. I'll update this issue when it is safe to remove the app setting. It's important to remove WEBSITE_USE_PLACEHOLDER=0 for better performance. Thanks!

**Comment by mhoeger** [comment]
The fix has been deployed as host version 2.0.12888. If you are on this version, please remove WEBSITE_USE_PLACEHOLDER=0 for better performance. Thanks!
